packages/core/src/utils/cursor-pagination/helpers.ts

```diff
@@ -0,0 +1,82 @@
+import { type SelectQueryBuilder, sql } from "kysely"
+import type { Cursor, Pageable } from "./types"
+
+type GetNextCursorOptions<T> = {
+  pageable: Pageable
+  buildCursor: (record: T) => Cursor
+}
+
+function getNextCursor<T>(records: T[], options: GetNextCursorOptions<T>): Cursor | null {
+  // We fetched take+1 records to determine if there is a record behind `take`.
+  // If we got exactly `take+1` records, there is a next page.
+  const hasNextPage = records.length === options.pageable.take + 1
+  if (!hasNextPage) {
+    return null
+  }
+
+  // biome-ignore lint/style/noNonNullAssertion: We know there is a record at -1 from the check above.
+  return options.buildCursor(records.at(-1)!)
+}
+
+type PaginatedQueryOptions<T> = {
+  pageable: Pageable
+  decodeCursor: (cursor: Cursor) => string
+  buildCursor: (record: T) => Cursor
```

Comment:
Suggest we call it encodeCursor and decodeCursor

Reply:
Seeing that it takes a record and not a cursor I didn't like encodeCursor. thoughts?